src/frontends/onnx/frontend/src/op/resize.cpp

```diff
@@ -145,8 +146,10 @@ OutputVector resize(const onnx_import::Node& node) {
 
     auto attrs = get_resize_attrs(node);
 
-    if (inputs.size() == 4)  // sizes input is provided
-    {
+    const auto use_sizes_input =
+        inputs.size() == 4 && std::dynamic_pointer_cast<NullNode>(inputs[3].get_node_shared_ptr()) == nullptr;
+
+    if (use_sizes_input) {
```

Comment:
We can use `is_null` helper
```suggestion
    if (inputs.size() == 4 && !is_null(inputs.at(3))
```

Reply:
Also it would be great to have a unit test for ONNX Resize op with NullNode `sizes` input.

Reply:
I've modified one of the test models. Setting an input to an empty string makes it optional and OV creates a `NullNode` for such inputs.